File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/CurriculumDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for course curriculum/syllabus.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CurriculumDTO {
    private List<CurriculumSectionDTO> sections;
    private String totalDuration; // e.g., "2.5 hours"
    private Integer totalDurationSeconds;
    private Integer totalLectures;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CurriculumDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for course curriculum/syllabus.
 */
data class CurriculumDTO(
    val sections: List<CurriculumSectionDTO>? = null,
    // e.g., "2.5 hours"
    val totalDuration: String? = null,
    val totalDurationSeconds: Int? = null,
    val totalLectures: Int? = null,
)
